src/main/scala/rps/Main.scala

```diff
@@ -59,4 +54,16 @@ object WebServer extends App with RouterDerivationModule {
   object Draw
 }
 
-case class Response (userMove: Move, computerMove: Move, result: Result)
\ No newline at end of file
+case class Response (userMove: Move, computerMove: Move, result: Result)
+
+trait ID {
+  type IdType
+  def generateGameId (game : Response) :IdType
+}
+
+object HashId extends ID {
```

Comment:
I think it would be more scala way if you implement this as a trait, and then in the class you have to use `generateGameId` you declare `with HashId`